### PR TITLE
Go 1.21

### PR DESCRIPTION
**Pull request by elcasteel**
# Description

Bump Go to 1.21.4 (version used by v0.7.0 of cloud-builders)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ci/cloudbuild/run-tests.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 steps:
 
-- name: 'gcr.io/$PROJECT_ID/prepare-go-workspace:0.6.5'
```
Comment (elcasteel) [inline ci/cloudbuild/run-tests.yaml:3]: replace with the released builders before merging

File: cloudbuild.yaml [hunk 2]
```diff
@@ -1,16 +1,16 @@
 steps:
 
-- name: 'gcr.io/cloud-builders/gcloud'
```
Comment (elcasteel) [inline cloudbuild.yaml:3]: revert this change before merging

**Comment by elcasteel** [comment]
Build-bot has run successfully while pointing to temporary cloud-builder images. When cloud-builder is released with go 1.21 we can update this PR and it will be ready to go

**Review by davidjumani — APPROVED** [review]

**Comment by elcasteel** [comment]
I can't approve a PR that I opened but LGTM

**Review by inFocus7 — APPROVED** [review]
